server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java

```diff
@@ -0,0 +1,338 @@
+    private static final String FIRST_ATTACHMENT = ATTACHMENTS + "[0]";
+    private static final String SECOND_ATTACHMENT = ATTACHMENTS + "[1]";
+
+
+    private final MainStepdefs mainStepdefs;
+    private final UserStepdefs userStepdefs;
+
+    private Response post;
+    private ValidatableResponse response;
+
+    @Inject
+    private GetMessagesMethodStepdefs(MainStepdefs mainStepdefs, UserStepdefs userStepdefs) {
+        this.mainStepdefs = mainStepdefs;
+        this.userStepdefs = userStepdefs;
+    }
+
+    @Given("^the user has a message in \"([^\"]*)\" mailbox with subject \"([^\"]*)\" and content \"([^\"]*)\"$")
+    public void appendMessage(String mailbox, String subject, String content) throws Throwable {
+        appendMessage(mailbox, Optional.empty(), subject, content, Optional.empty());
+    }
+
+    @Given("^the user has a message in \"([^\"]*)\" mailbox with content-type \"([^\"]*)\" subject \"([^\"]*)\" and content \"([^\"]*)\"$")
+    public void appendMessage(String mailbox, String contentType, String subject, String content) throws Throwable {
+        appendMessage(mailbox, Optional.of(contentType), subject, content, Optional.empty());
+    }
+
+    @Given("^the user has a message in \"([^\"]*)\" mailbox with subject \"([^\"]*)\" and content \"([^\"]*)\" with headers$")
+    public void appendMessage(String mailbox, String subject, String content, DataTable headers) throws Throwable {
+        appendMessage(mailbox, Optional.empty(), subject, content, Optional.of(headers.asMap(String.class, String.class)));
+    }
+
+    private void appendMessage(String mailbox, Optional<String> contentType, String subject, String content, Optional<Map<String, String>> headers) throws Exception {
+        ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
+        mainStepdefs.jmapServer.serverProbe().appendMessage(userStepdefs.username, 
+                new MailboxPath(MailboxConstants.USER_NAMESPACE, userStepdefs.username, mailbox),
+                new ByteArrayInputStream(message(contentType, subject, content, headers).getBytes()), 
+                Date.from(dateTime.toInstant()), false, new Flags());
+    }
+
+    private String message(Optional<String> contentType, String subject, String content, Optional<Map<String,String>> headers) {
+        Function<Set<Map.Entry<String, String>>, String> entriesToString = entries -> entries.stream()
+                .map(this::entryToPair)
+                .map(this::joinKeyValue)
+                .collect(Collectors.joining("\r\n", "", "\r\n"));
+
+        String headersAsString = headers
+                .map(map -> map.entrySet())
+                .map(entriesToString)
+                .orElse("");
+                
+        String contentTypeAsString = contentType
+                .map(value -> "Content-Type: " + value + "\r\n")
+                .orElse("");
+
+        return headersAsString + contentTypeAsString + "Subject: " + subject + "\r\n\r\n" + content;
```

Comment:
Then, we should modify then appendMesssage method in order to take a mime4j Message in replacement of all parameters.